core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleDecommissionViaCMHandler.java

```diff
@@ -173,6 +177,19 @@ protected Selectable doAccept(HandlerEvent<StopStartDownscaleDecommissionViaCMRe
         }
     }
 
+    private void excludeAndLogDisconnectedNMsForDownscale(Stack stack, Map<String, InstanceMetadataView> hostsToRemove,
+            StopStartDownscaleDecommissionViaCMRequest request) {
+        ClusterHealthService clusterHealthService = clusterApiConnectors.getConnector(stack).clusterHealthService();
+        Set<String> disconnectedNMHostNames = clusterHealthService.getDisconnectedNodeManagers();
+        if (!disconnectedNMHostNames.isEmpty()) {
+            LOGGER.info("Found {} disconnected NodeManagers: {}, for hostgroup {}. Excluding them from de-commission list", disconnectedNMHostNames.size(),
+                    disconnectedNMHostNames, request.getHostGroupName());
+            flowMessageService.fireEventAndLog(stack.getId(), UPDATE_IN_PROGRESS.name(), CLUSTER_SCALING_STOPSTART_DOWNSCALE_EXCLUDE_LOST_NODES,
+                    String.valueOf(disconnectedNMHostNames.size()), String.join(", ", disconnectedNMHostNames));
+            hostsToRemove.entrySet().removeIf(e -> disconnectedNMHostNames.contains(StringUtils.lowerCase(e.getKey())));
```

Comment:
Out of curiosity, what benefit does `StringUtils.lowerCase()` have vs `String.toLowerCase()`?

Reply:
Here, The input will be static content and String.toLowerCase() is a non-static method. So, StringUtils.lowerCase() is used.

Reply:
Let me ask differently:
```
e.getKey().toLowerCase()
```
vs
```
StringUtils.lowerCase(e.getKey())
```

As I see, `StringUtils.lowerCase()` is implemented this way:
```
    public static String lowerCase(final String str) {
        if (str == null) {
            return null;
        }
        return str.toLowerCase();
    }
```
So the only difference here is the latter being null-safe, whereas `e.getKey().toLowerCase()` will obviously throw an NPE if any key in `hostsToRemove` is `null`. Is this something we expect to be prepared for? :smile:

Reply:
Anyway, this is only a subtlety, so no need to change it.